File: PythonAlgorithmInterView/선형자료구조/팰린드롬연결리스트.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {boolean}
 * 234번 문제.
 */
 var isPalindrome = function(head) {
    q = [];

    if( head.length === 0) return true;

    for (let index = 0; index < head.length; index++) {
        q.push(head[index].val)
    }


    while(q.length > 1){
        if(q.shift() != q.pop()){
            return false;
        }
    }

    return true;
};

let head =[
    {
        val : 1,
        next : 'next'
    },
    {
        val : 2,
        next : 'next'
    }
]

console.log(isPalindrome(head))

File: PythonAlgorithmInterView/선형자료구조/팰린드롬연결리스트.py

def isPalindrome(self, head: [list]) -> bool:
    q : list = []

    if not head:
        return True

    node = head

    while node is not None:
        q.append(node.val)
        node = node.next

    while len(q) > 1:
        if q.pop(0) != q.pop() :
            return False

    return True

    # while left < right :
    #     head.p

if __name__ == '__main__':
    head = [1, 2, 2, 1]

    print(isPalindrome(None, head))